crates/curp/src/server/raw_curp/log.rs

```diff
@@ -113,19 +120,51 @@ impl<C: Command> LogEntryVecDeque<C> {
     fn push_back(&mut self, entry: Arc<LogEntry<C>>) -> Result<(), bincode::Error> {
         let entry_size = serialized_size(&entry)?;
 
+        if entry_size > self.batch_limit {
+            warn!("entry_size of an entry > batch_limit, which may be too small.",);
+        }
+
         self.entries.push_back(entry);
-        let Some(&pre_entries_size) = self.batch_index.back() else {
-            unreachable!("batch_index cannot be None")
-        };
-        self.batch_index
-            .push_back(pre_entries_size.overflow_add(entry_size));
+        self.entry_size.push_back(entry_size);
+        self.batch_index.push_back(0); // placeholder
+        self.last_batch_size += entry_size;
+
+        while self.last_batch_size > self.batch_limit {
```

Comment:
Why do you skip the equal case here?  I think we should update the `batch_index` when the `self.last_batch_size == self.batch_limit`.

Reply:
The reason for the design is to unify the right index of a range, which are different between `self.last_batch_size > self.batch_limit` and `self.last_batch_size == self.batch_limit`. But I think your explanation makes sense, so I‘ve changed my code.